### DISCUSSION TITLE
File Security

### DISCUSSION THREAD

**Discussion by MPatriot** [oqtane/oqtane.framework#3228] [Q&A] [answered]
How would one protect a file, whether it be a document, video, audio spreadsheet whatever from unauthorized access?

This is probably obvious but off the top of my head I can't think of a safe way to do this with  an Oqtane app all alone.  It seems to me that everything, including directories like the content directory are wide open to whomever has the url.  

Am I missing something or would I need to secure files in another location, such as Google dos or other secure file server and then programmatically fetch them based on saved permissions of the user?

As always, I appreciate the Oqtane contributors and efforts to keep up in a faster and faster changing environment.

Mark

**Comment by sbwalker** [comment, marked as answer]
@MPatriot Oqtane supports 2 types of Folders - private and public. 

Private folders store their files outside of the wwwroot so that the only way they can be accessed is via the Oqtane API (where Folder permissions are used to control access). The default root folder created for each site is a private folder. 

Public folders store their files inside the wwwroot so they can be accessed as long as you know the specific url path.

Folders can be created/managed in the File Manager in the Admin Dashboard. Or they can be managed programmatically using the FolderService API.

**Reply by MPatriot** [reply]
Thanks @sbwalker

I knew I knew that! Haha.  After the struggles I had not to long ago with the webservers permissions I wasn't sure what to think anymore.

After looking at it again, I see and remember now that you can be very specific just using the Content folder provided by Oqtane.  I especially like that you can create a directory with no permissions granted (though Host User would always have full access) and then add permissions as granular as individual users.

This was tested by simply creating a test directory and uploading a file to it.  When clicking the link I could view the file on a different page but after logging out, once I refreshed the file page it wouldn't show up.

How about that!  It works exactly as originally expected!  No problems to report this time:)

Thanks,

Mark
